tests/pyfunc/test_model_export_with_class_and_artifacts.py

```diff
@@ -1443,3 +1443,37 @@ def test_predict(sk_model, model_input):
 
     artifacts = _load_pyfunc(pyfunc_model_path).context.artifacts
     assert all("\\" not in artifact_uri for artifact_uri in artifacts.values())
+
+
+@pytest.mark.large
+def test_multiple_artifacts_with_same_artifact_relative_path(
+    sklearn_knn_model, model_path, iris_data
+):
+    default_model_path = "model"
+    with mlflow.start_run():
+        model_info = mlflow.sklearn.log_model(
+            sk_model=sklearn_knn_model, artifact_path=default_model_path
+        )
+        first_model_uri = model_info.model_uri
+
+    with mlflow.start_run():
+        model_info = mlflow.sklearn.log_model(
+            sk_model=sklearn_knn_model, artifact_path=default_model_path
+        )
+        second_model_uri = model_info.model_uri
+
+    class ArtifactValidationModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input):
+            first_model_resolved_path = context.artifacts["first_model"]
```

Comment:
This is a great change. :) When ready to move this from draft to review, could an explanation of this functionality be added to https://www.mlflow.org/docs/latest/models.html#custom-python-models to demonstrate to users how to use this? 
It's very intuitive as-is; just more for raising awareness of how to use the context artifact reference in this manner will help to encourage usage like this :)

Reply:
@BenWilson2 I added the documentation you mentioned. Please let me know what you think about it!